pkg/registry/core/replicationcontroller/strategy.go

```diff
@@ -125,14 +128,31 @@ func (rcStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object)
 func (rcStrategy) Validate(ctx context.Context, obj runtime.Object) field.ErrorList {
 	controller := obj.(*api.ReplicationController)
 	opts := pod.GetValidationOptionsFromPodTemplate(controller.Spec.Template, nil)
+
+	// Run imperative validation
 	allErrs := corevalidation.ValidateReplicationController(controller, opts)
+
+	// If DeclarativeValidation feature gate is enabled, run both validations
 	if utilfeature.DefaultFeatureGate.Enabled(features.DeclarativeValidation) {
-		declarativeErrs := rest.ValidateDeclaratively(ctx, nil, legacyscheme.Scheme, obj)
+		// Run declarative validation
+		var declarativeErrs field.ErrorList
+		func() {
+			defer func() {
+				if r := recover(); r != nil {
+					declarativeErrs = append(declarativeErrs, field.InternalError(nil, fmt.Errorf("panic during declarative validation: %v", r)))
```

Comment:
Is this logged, too?

Do we want a metric for how often this happens?

Reply:
I have updated this so that there is now a new meteri - declarative_validation_panic_total and also logged this.  Additionally I updated this initial logic such that.  

if DeclarativeValidationTakeover=enabled and ValidateDeclaratively panics -> output a validation error (append to the ErroList) stating the panic

if DeclarativeValidatoinTakeover=disabled and ValidateDeclaratively panics -> log a warning that the panic occured but don't modify the ErrorList in any way